src/framing/codec.rs

```diff
@@ -40,6 +42,28 @@ impl Codec for MLSPlaintext {
     }
 }
 
+impl MLSPlaintext {
+    /// Decode an `MLSPlaintext` with ciphersuite and protocol version information.
+    /// This should be used instead of the raw decoding function in order to
+    /// update the `MLSPlaintext` with the missing information.
+    pub fn decode_with_context(
```

Comment:
Yeah, this is a spec issue imo. MlsPlaintext isn't well defined if you want to be as defensive as we are here now. But this is not for the library to decide here but for consumers. If there's a consumer that needs a different API we can think about how to make it better. Right now it doesn't look like there is one.